sharding/contracts/sharding_manager_test.go

```diff
@@ -16,412 +16,320 @@ import (
 	"github.com/ethereum/go-ethereum/sharding"
 )
 
-type SMCConfig struct {
-	notaryLockupLenght   *big.Int
-	proposerLockupLength *big.Int
+type testAccount struct {
+	addr    common.Address
+	privKey *ecdsa.PrivateKey
+	txOpts  *bind.TransactOpts
 }
 
+type testAcconts []testAccount
+
 var (
 	mainKey, _                   = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
 	accountBalance2000Eth, _     = new(big.Int).SetString("2000000000000000000000", 10)
 	notaryDepositInsufficient, _ = new(big.Int).SetString("999000000000000000000", 10)
 	notaryDeposit, _             = new(big.Int).SetString("1000000000000000000000", 10)
 	FastForward100Blocks         = 100
-	// smcConfig                    = SMCConfig{
-	// 	notaryLockupLenght:   new(big.Int).SetInt64(1),
-	// 	proposerLockupLength: new(big.Int).SetInt64(1),
-	// }
-	ctx = context.Background()
+	ctx                          = context.Background()
 )
 
+// fastForward is a helper function to skip through n period.
+func fastForward(backend *backends.SimulatedBackend, p int) {
+	for i := 0; i < p*int(sharding.PeriodLength); i++ {
+		backend.Commit()
+	}
+}
+
+// initAccounts is a helper function to initialize backend with the n accounts,
+// returns an array of testAccount struct.
+func initAccounts(n int) (testAcconts, map[common.Address]core.GenesisAccount) {
+	genesis := make(core.GenesisAlloc)
+	var testAccounts []testAccount
+
+	for i := 0; i < n; i++ {
+		privKey, _ := crypto.GenerateKey()
+		addr := crypto.PubkeyToAddress(privKey.PublicKey)
+		txOpts := bind.NewKeyedTransactor(privKey)
+		testAccounts = append(testAccounts, testAccount{addr, privKey, txOpts})
+		genesis[addr] = core.GenesisAccount{
+			Balance: accountBalance2000Eth,
+		}
+	}
+	return testAccounts, genesis
+}
+
 // deploySMCContract is a helper function for deploying SMC.
 func deploySMCContract(backend *backends.SimulatedBackend, key *ecdsa.PrivateKey) (common.Address, *types.Transaction, *SMC, error) {
 	transactOpts := bind.NewKeyedTransactor(key)
 	defer backend.Commit()
 	return DeploySMC(transactOpts, backend)
 }
 
-// TestContractCreation tests SMC smart contract can successfully be deployed.
-func TestContractCreation(t *testing.T) {
-	addr := crypto.PubkeyToAddress(mainKey.PublicKey)
-	backend := backends.NewSimulatedBackend(core.GenesisAlloc{addr: {Balance: accountBalance2000Eth}})
-	_, _, _, err := deploySMCContract(backend, mainKey)
-	backend.Commit()
-	if err != nil {
-		t.Fatalf("can't deploy SMC: %v", err)
+// registerNotaries is a helper function register notaries in batch,
+// it also verifies notary registration is successful.
+func (a testAcconts) registerNotaries(t *testing.T, backend *backends.SimulatedBackend, smc *SMC, deposit *big.Int, params ...int) error {
```

Comment:
I don't think should accept testing.T, it should just return an error and the test can handle it